compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -12972,52 +12972,85 @@ private STNode parseConditionalExpression(STNode lhsExpr) {
         // parsing conditional expr. That is done by lowering the current precedence.
         STNode middleExpr = parseExpression(OperatorPrecedence.ANON_FUNC_OR_LET, true, false, true);
 
+        // There are special cases when the nextToken is not `:`. Any expression that contains qualified name ref at
+        // the beginning or at the end, is a possible match for conditionalExpr.
         if (peek().kind != SyntaxKind.COLON_TOKEN) {
-            // There are special cases when the nextToken is not `:`. Any expression that contains qualified name ref at
-            // the beginning or at the end, is a possible match for conditionalExpr.
+            if (middleExpr.kind == SyntaxKind.CONDITIONAL_EXPRESSION) {
```

Comment:
Can't we simply do this by passing `isInConditionalExpr` argument to `parseConditionalExpression`

Reply:
No help from `isInConditionalExpr` here. 
But we can use it to replace the 
`boolean isInConditionalExpr = this.errorHandler.getParentContext() == ParserRuleContext.CONDITIONAL_EXPRESSION;`

Updated with b737fab